edweiss-app/app/(app)/lectures/slides/index.tsx

```diff
@@ -276,50 +277,70 @@ const LectureViewer: React.FC<{
     setCurrentPage: (currentPage: number) => void;
     page: number;
     isLandscape: boolean;
-}> = ({ uri, widthPorp, heightProp, currentEvent, currentQuestion, setNumPages, setCurrentPage, page, isLandscape }) => {
-
-    return (currentEvent && currentEvent.type === "invalid") ? (
-        <Pdf
-            trustAllCerts={false}
-            source={{ uri }}
-            renderActivityIndicator={() => <ActivityIndicator size="large" />}
-            enablePaging
-            onLoadComplete={(totalPages) => setNumPages(totalPages)}
-            onPageChanged={(currentPage) => setCurrentPage(currentPage)}
-            onError={(error) => console.log(error)}
-            page={page}
-            horizontal
-            style={{
-                flex: 1,
-                width: Dimensions.get('window').width * widthPorp,
-                height: Dimensions.get('window').height * heightProp,
-            }}
-        />
-    ) : (
-
-        currentQuestion && <>
-            <TView justifyContent='center' alignItems='center' mt='lg' mb={isLandscape ? 'sm' : 'xs'}>
-                <TText bold size='lg' mb='sm'>{t('showtime:question_broadcast_ans_title')}</TText>
-            </TView>
+    courseId: string;
+    lectureId: string;
+}> = ({ uri, widthPorp, heightProp, currentEvent, currentQuestion, setNumPages, setCurrentPage, page, isLandscape, courseId, lectureId }) => {
+
+    const isQuiz = currentEvent && currentEvent.type === "quiz";
+    const isQuestion = currentEvent && currentEvent.type === "question" && currentQuestion;
+    let toBeDisplayed;
```

Comment:
I do not know if using a let for component display is the most optimal way to do this.. You could just do a some ternary functions in the return I think

Reply:
Indeed. If you go in the git history, you can actually see that I was using them actually.
However, sonar cloud was actually complaining about it. And enforced me to use a `let` and `if-elseif-else` structure instead in order to reduce code complexity and to extract stuff in external variables.
That's how I got 0 new issues 😁.

Reply:
Oh I understand if thats the case all good, my mistake.